### ISSUE TITLE
cargo fix does not honor .gitignore 

### ISSUE THREAD

**Issue by eminence** [rust-lang/cargo#5771]
(unclear if this issue should be filed in this repo, or the rustfix repo)

`cargo fix` does not honor a `.gitignore` when doing dirty detection, which means that you have to run `git clean` in between invocations. 

    $ cargo new --lib testlib && cd testlib

    $ git add -A
    $ git commit -m "initial commit"
    [master (root-commit) d00917c] initial commit
     3 files changed, 16 insertions(+)
     create mode 100644 .gitignore
     create mode 100644 Cargo.toml
     create mode 100644 src/lib.rs

```
$ cargo fix
error: the working directory of this project is detected as dirty, and `cargo fix` can potentially perform destructive changes; if you'd like to suppress this error pass `--allow-dirty`, or commit the changes to these files:

  * Cargo.lock
  * target/
```

```
$ git status                                                                                                                                       
On branch master
nothing to commit, working tree clean
```

```
$ cat .gitignore 
/target
**/*.rs.bk
Cargo.lock
```

cargo 1.29.0-nightly (506eea76e 2018-07-17)

**Comment by dwijnand** [comment]
Hey @eminence, looks like @alexcrichton has got you covered and is fixing this in https://github.com/rust-lang/cargo/pull/5770 (pending merge).

**Comment by killercup** [comment]
Can you confirm this is fixed by #5770? (You need to build cargo yourself)

**Comment by eminence** [comment]
Confirmed!  it is fixed!
